File: Java/Runtime/ReflectionTest/Program1.java
record Student(String name, String course, float score) {}

record Teacher(String name, int experience) {}

class Program {
	
	private static void printAsXml(Student info) {
		System.out.println("<Student>");
		System.out.printf("  <name>%s</name>%n", info.name());
		System.out.printf("  <course>%s</course>%n", info.course());
		System.out.printf("  <score>%s</score>%n", info.score());
		System.out.println("</Student>");
		System.out.println();
	}
	
	private static void printAsXml(Teacher info) {
		System.out.println("<Teacher>");
		System.out.printf("  <name>%s</name>%n", info.name());
		System.out.printf("  <experience>%s</experience>%n", info.experience());
		System.out.println("</Teacher>");
		System.out.println();
	}

	public static void main(String[] args) {
		printAsXml(new Student("Jill", "C++", 83.5f));
		printAsXml(new Student("Jack", "Java", 68.0f));
		printAsXml(new Teacher("John", 9));
	}
}

File: DotNet/Runtime/ReflectionTest/DemoApp/Program1.cs
record Student (string Name, string Course, float Score);

record Teacher (string Name, int Experience);

class Program
{
    private static void PrintAsXml(object info)
    {
        Console.WriteLine("<Student>");
        Console.WriteLine("  <Name>{0}</Name>", info.Name);
        Console.WriteLine("  <Course>{0}</Course>", info.Course);
        Console.WriteLine("  <Score>{0}</Score>", info.Score);
        Console.WriteLine("</Student>");
        Console.WriteLine();
    }

    private static void PrintAsXml(Teacher info)
    {
        Console.WriteLine("<Teacher>");
        Console.WriteLine("  <Name>{0}</Name>", info.Name);
        Console.WriteLine("  <Experience>{0}</Experience>", info.Experience);
        Console.WriteLine("</Teacher>");
        Console.WriteLine();
    }


    public static void Main()
    {
        PrintAsXml(new Student("Jack", "Java", 67));
        PrintAsXml(new Teacher("Jill", 9));
    }
}